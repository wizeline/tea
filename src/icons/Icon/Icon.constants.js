const viewBox = '0 0 24 24';
const preserveAspectRatio = 'xMidYMid meet';

export {
  viewBox,
  preserveAspectRatio,
};
