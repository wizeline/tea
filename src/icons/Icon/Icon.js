// @flow

import React from 'react';
import CircleIcon from '../CircleIcon/CircleIcon';
import SVGIcon from '../SVGIcon/SVGIcon';
import type { IconPropTypes } from './IconPropTypes';
import getChildProps from './IconHelpers';
import getSVG from '../SVGIcon/data';
import { warn } from '../../utils';
import defaultIconTheme from '../../themes/defaultTheme/components/Icon';

const getInvalidIconNameWarningMessage = (name: string) =>
  `An invalid \`name\` prop with value \`${name}\` was supplied to <Icon />.`;

const getSizeProps = ({ size }) => {
  switch (size) {
    case 'xsmall':
      return { xmall: true };
    case 'medium':
      return { medium: true };
    case 'large':
      return { large: true };
    case 'xlarge':
      return { xlarge: true };
    default:
      return { small: true };
  }
};

const Icon = (iconProps: IconPropTypes) => {
  const { name, round } = iconProps;
  const svg = getSVG(name);
  if (!svg) {
    const warningMessage = getInvalidIconNameWarningMessage(name);
    return warn(warningMessage);
  }
  const childProps = { ...getChildProps(iconProps), svgData: svg };
  const sizeProps = getSizeProps(childProps);
  return round ? (
    <CircleIcon {...childProps} {...sizeProps} />
  ) : (
    <SVGIcon {...childProps} {...sizeProps} />
  );
};

Icon.defaultProps = {
  theme: defaultIconTheme,
};

export default Icon;
