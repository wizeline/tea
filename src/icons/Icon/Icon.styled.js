import styled from 'styled-components';
import colors from '../../styles/themes/colors';

const defaultIconSize = '16px';

const IconDivStyled = styled.div`
  position: relative;
  height: ${ props => props.size || defaultIconSize };
  width: ${ props => props.size || defaultIconSize };
`;

const IconSvgStyled = styled.svg`
  fill: ${ colors.steelGrey };
  user-select: none;
  &:hover {
    fill: ${ colors.darkSkyBlue };
  }
`;

export { IconDivStyled, IconSvgStyled };
