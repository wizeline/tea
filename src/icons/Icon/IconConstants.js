// @flow

const multipleTruthyKeyError =
  'ERROR: Multiple keys associated with a truthy value, offending keys:';

export default multipleTruthyKeyError;
