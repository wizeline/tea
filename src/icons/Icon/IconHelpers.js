// @flow

/* eslint flowtype-errors/enforce-min-coverage: 0 */

import pick from 'ramda/src/pick';
import pipe from 'ramda/src/pipe';
import propKeys from './IconConstants';
import { getTruthyKey, pickSizes } from '../../utils/coreUtils';
import type { IconPropTypes } from './IconPropTypes';

const getSize = pipe(pickSizes, getTruthyKey);

const getChildProps = (iconProps: IconPropTypes) => {
  const { NAME, COLOR, BACKGROUND } = propKeys;
  const { name } = pick([NAME], iconProps);
  const size = getSize(iconProps);
  const colors = pick([COLOR, BACKGROUND], iconProps);
  return { name, size, ...colors };
};

export default getChildProps;
