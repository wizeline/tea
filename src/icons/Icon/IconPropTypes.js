// @flow

export type IconPropTypes = {
  name: string,
  round?: boolean,
  // Size Props
  xsmall?: boolean,
  small?: boolean,
  medium?: boolean,
  large?: boolean,
  xlarge?: boolean,
  // Color Props
  color?: string,
  background?: string,
};
