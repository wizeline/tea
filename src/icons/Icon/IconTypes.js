// @flow

export type IconPropTypes = {
  name: string,
  round?: boolean,
  // ...sizeProps,
  xsmall?: boolean,
  small?: boolean,
  medium?: boolean,
  large?: boolean,
  xlarge?: boolean,
  // ...colorProps,
  color?: string,
  background?: string,
};
