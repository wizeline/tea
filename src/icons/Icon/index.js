// @flow

/* eslint flowtype-errors/enforce-min-coverage: 0 */

import React from 'react';
import withTheme from '../../utils/withTheme';
import Icon from './Icon';

export default withTheme(Icon);
