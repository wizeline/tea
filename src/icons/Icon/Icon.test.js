import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Icon from './Icon';
import {
  shallowExpectInvalid,
  shallowExpectToMatchSnapShot,
  mountExpectToThrow,
} from '../../utils/testHelpers';

describe('Icon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Icon name="cross" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders correctly without defined size or background', () => {
    shallowExpectToMatchSnapShot(<Icon round name="cross" />);
  });
  it('renders correctly with round, large and background', () => {
    shallowExpectToMatchSnapShot(
      <Icon round large name="cross" background="yellow" />,
    );
  });
  it('renders correctly with round, large and undefined background', () => {
    shallowExpectToMatchSnapShot(<Icon round large name="cross" />);
  });
  it('renders correctly, invalid size ignored', () => {
    shallowExpectToMatchSnapShot(<Icon round xlarge name="cross" />);
  });
  it('throws an error with multiple truthy sizes', () => {
    mountExpectToThrow(<Icon round xlarge large name="cross" />);
  });
  it('renders nothing with an invalid name and undefined size attribute', () => {
    shallowExpectInvalid(<Icon name="InvalidIconName" />);
  });
  it('renders nothing with invalid name and invalid size', () => {
    shallowExpectInvalid(<Icon name="InvalidIconName" size="InvalidSize" />);
  });
  it('renders nothing with an undefined name and undefined size', () => {
    shallowExpectInvalid(<Icon />);
  });
  it('renders nothing with an invalid name and undefined size attribute', () => {
    shallowExpectInvalid(<Icon round name="InvalidIconName" />);
  });
  it('renders nothing with round, invalid name and invalid size', () => {
    shallowExpectInvalid(
      <Icon round name="InvalidIconName" size="InvalidSize" />,
    );
  });
});
