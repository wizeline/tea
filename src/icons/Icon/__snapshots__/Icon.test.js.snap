// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<SVGIcon
  svgData={
    <svg
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(1 1)"
      >
        <path
          d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
        />
      </g>
    </svg>
  }
  theme={
    Object {
      "backgroundDanger": "#ED524C",
      "backgroundDangerDisabled": "#E6E7E8",
      "backgroundDangerHover": "#F06D68",
      "backgroundInput": "#FFFFFF",
      "backgroundPrimary": "#50A5E7",
      "backgroundPrimaryDisabled": "#E6E7E8",
      "backgroundPrimaryHover": "#6BB3EB",
      "backgroundSecondary": "#FFFFFF",
      "backgroundSecondaryDisabled": "#FFFFFF",
      "backgroundSecondaryHover": "#FFFFFF",
      "textBody": "#3F484B",
      "textDanger": "#FFFFFF",
      "textDangerDisabled": "#CDD0D2",
      "textDangerHover": "#FFFFFF",
      "textHeading": "#3F484B",
      "textIcon": "#FFFFFF",
      "textIconInvert": "#222729",
      "textInput": "#3F484B",
      "textInputFocus": "#8B9193",
      "textInputInvalid": "#E92720",
      "textInputPlaceholder": "#8B9193",
      "textPrimary": "#3F484B",
      "textPrimaryDisabled": "#8B9193",
      "textPrimaryHover": "#FFFFFF",
      "textPrimaryInvert": "#FFFFFF",
      "textSecondary": "#3F484B",
      "textSecondaryDisabled": "#CDD0D2",
      "textSecondaryHover": "#FFFFFF",
      "textSubtitle": "#8B9193",
    }
  }
/>
`;

exports[`Icon renders correctly with round, large and background 1`] = `
<CircleIcon
  background="yellow"
  size="large"
  svgData={
    <svg
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(1 1)"
      >
        <path
          d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
        />
      </g>
    </svg>
  }
  theme={
    Object {
      "backgroundDanger": "#ED524C",
      "backgroundDangerDisabled": "#E6E7E8",
      "backgroundDangerHover": "#F06D68",
      "backgroundInput": "#FFFFFF",
      "backgroundPrimary": "#50A5E7",
      "backgroundPrimaryDisabled": "#E6E7E8",
      "backgroundPrimaryHover": "#6BB3EB",
      "backgroundSecondary": "#FFFFFF",
      "backgroundSecondaryDisabled": "#FFFFFF",
      "backgroundSecondaryHover": "#FFFFFF",
      "textBody": "#3F484B",
      "textDanger": "#FFFFFF",
      "textDangerDisabled": "#CDD0D2",
      "textDangerHover": "#FFFFFF",
      "textHeading": "#3F484B",
      "textIcon": "#FFFFFF",
      "textIconInvert": "#222729",
      "textInput": "#3F484B",
      "textInputFocus": "#8B9193",
      "textInputInvalid": "#E92720",
      "textInputPlaceholder": "#8B9193",
      "textPrimary": "#3F484B",
      "textPrimaryDisabled": "#8B9193",
      "textPrimaryHover": "#FFFFFF",
      "textPrimaryInvert": "#FFFFFF",
      "textSecondary": "#3F484B",
      "textSecondaryDisabled": "#CDD0D2",
      "textSecondaryHover": "#FFFFFF",
      "textSubtitle": "#8B9193",
    }
  }
/>
`;

exports[`Icon renders correctly with round, large and undefined background 1`] = `
<CircleIcon
  size="large"
  svgData={
    <svg
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(1 1)"
      >
        <path
          d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
        />
      </g>
    </svg>
  }
  theme={
    Object {
      "backgroundDanger": "#ED524C",
      "backgroundDangerDisabled": "#E6E7E8",
      "backgroundDangerHover": "#F06D68",
      "backgroundInput": "#FFFFFF",
      "backgroundPrimary": "#50A5E7",
      "backgroundPrimaryDisabled": "#E6E7E8",
      "backgroundPrimaryHover": "#6BB3EB",
      "backgroundSecondary": "#FFFFFF",
      "backgroundSecondaryDisabled": "#FFFFFF",
      "backgroundSecondaryHover": "#FFFFFF",
      "textBody": "#3F484B",
      "textDanger": "#FFFFFF",
      "textDangerDisabled": "#CDD0D2",
      "textDangerHover": "#FFFFFF",
      "textHeading": "#3F484B",
      "textIcon": "#FFFFFF",
      "textIconInvert": "#222729",
      "textInput": "#3F484B",
      "textInputFocus": "#8B9193",
      "textInputInvalid": "#E92720",
      "textInputPlaceholder": "#8B9193",
      "textPrimary": "#3F484B",
      "textPrimaryDisabled": "#8B9193",
      "textPrimaryHover": "#FFFFFF",
      "textPrimaryInvert": "#FFFFFF",
      "textSecondary": "#3F484B",
      "textSecondaryDisabled": "#CDD0D2",
      "textSecondaryHover": "#FFFFFF",
      "textSubtitle": "#8B9193",
    }
  }
/>
`;

exports[`Icon renders correctly without defined size or background 1`] = `
<CircleIcon
  svgData={
    <svg
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(1 1)"
      >
        <path
          d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
        />
      </g>
    </svg>
  }
  theme={
    Object {
      "backgroundDanger": "#ED524C",
      "backgroundDangerDisabled": "#E6E7E8",
      "backgroundDangerHover": "#F06D68",
      "backgroundInput": "#FFFFFF",
      "backgroundPrimary": "#50A5E7",
      "backgroundPrimaryDisabled": "#E6E7E8",
      "backgroundPrimaryHover": "#6BB3EB",
      "backgroundSecondary": "#FFFFFF",
      "backgroundSecondaryDisabled": "#FFFFFF",
      "backgroundSecondaryHover": "#FFFFFF",
      "textBody": "#3F484B",
      "textDanger": "#FFFFFF",
      "textDangerDisabled": "#CDD0D2",
      "textDangerHover": "#FFFFFF",
      "textHeading": "#3F484B",
      "textIcon": "#FFFFFF",
      "textIconInvert": "#222729",
      "textInput": "#3F484B",
      "textInputFocus": "#8B9193",
      "textInputInvalid": "#E92720",
      "textInputPlaceholder": "#8B9193",
      "textPrimary": "#3F484B",
      "textPrimaryDisabled": "#8B9193",
      "textPrimaryHover": "#FFFFFF",
      "textPrimaryInvert": "#FFFFFF",
      "textSecondary": "#3F484B",
      "textSecondaryDisabled": "#CDD0D2",
      "textSecondaryHover": "#FFFFFF",
      "textSubtitle": "#8B9193",
    }
  }
/>
`;

exports[`Icon renders correctly, invalid size ignored 1`] = `
<CircleIcon
  size="xlarge"
  svgData={
    <svg
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(1 1)"
      >
        <path
          d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
        />
      </g>
    </svg>
  }
  theme={
    Object {
      "backgroundDanger": "#ED524C",
      "backgroundDangerDisabled": "#E6E7E8",
      "backgroundDangerHover": "#F06D68",
      "backgroundInput": "#FFFFFF",
      "backgroundPrimary": "#50A5E7",
      "backgroundPrimaryDisabled": "#E6E7E8",
      "backgroundPrimaryHover": "#6BB3EB",
      "backgroundSecondary": "#FFFFFF",
      "backgroundSecondaryDisabled": "#FFFFFF",
      "backgroundSecondaryHover": "#FFFFFF",
      "textBody": "#3F484B",
      "textDanger": "#FFFFFF",
      "textDangerDisabled": "#CDD0D2",
      "textDangerHover": "#FFFFFF",
      "textHeading": "#3F484B",
      "textIcon": "#FFFFFF",
      "textIconInvert": "#222729",
      "textInput": "#3F484B",
      "textInputFocus": "#8B9193",
      "textInputInvalid": "#E92720",
      "textInputPlaceholder": "#8B9193",
      "textPrimary": "#3F484B",
      "textPrimaryDisabled": "#8B9193",
      "textPrimaryHover": "#FFFFFF",
      "textPrimaryInvert": "#FFFFFF",
      "textSecondary": "#3F484B",
      "textSecondaryDisabled": "#CDD0D2",
      "textSecondaryHover": "#FFFFFF",
      "textSubtitle": "#8B9193",
    }
  }
/>
`;

exports[`Icon renders nothing with an invalid name and undefined size attribute 1`] = `""`;

exports[`Icon renders nothing with an invalid name and undefined size attribute 2`] = `""`;

exports[`Icon renders nothing with an undefined name and undefined size 1`] = `""`;

exports[`Icon renders nothing with invalid name and invalid size 1`] = `""`;

exports[`Icon renders nothing with round, invalid name and invalid size 1`] = `""`;
